contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -231,29 +231,29 @@ contract ItemsFacet is Modifiers {
         Aavegotchi storage aavegotchi = s.aavegotchis[_tokenId];
         require(aavegotchi.status == LibAavegotchi.STATUS_AAVEGOTCHI, "LibAavegotchi: Only valid for AG");
         emit EquipWearables(_tokenId, aavegotchi.equippedWearables, _wearablesToEquip);
-        emit EquipDelegatedWearables(_tokenId, s.gotchiEquippedItemsInfo[_tokenId].equippedCommitmentIds, _commitmentIdsToEquip);
-        GotchiEquippedCommitmentsInfo storage _gotchiInfo = s.gotchiEquippedItemsInfo[_tokenId];
+        emit EquipDelegatedWearables(_tokenId, s.gotchiEquippedDepositsInfo[_tokenId].equippedDepositIds, _depositIdsToEquip);
+        GotchiEquippedDepositsInfo storage _gotchiInfo = s.gotchiEquippedDepositsInfo[_tokenId];
 
         for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
             
             uint256 toEquipId = _wearablesToEquip[slot];
             uint256 existingEquippedWearableId = aavegotchi.equippedWearables[slot];
-            bool _sameWearablesIds = toEquipId == existingEquippedWearableId;
 
-            uint256 _commitmentIdToEquip = _commitmentIdsToEquip[slot];
-            uint256 _existingEquippedCommitmentId = _gotchiInfo.equippedCommitmentIds[slot];
+            uint256 _depositIdToEquip = _depositIdsToEquip[slot];
+            uint256 _existingEquippedDepositId = _gotchiInfo.equippedDepositIds[slot];
 
-            //If the new wearable value is equal to the current equipped wearable in that slot
+            //If the new wearable value is equal to the current equipped wearable in that slot and the depositId is the same
             //do nothing
-            if (_sameWearablesIds && _existingEquippedCommitmentId == _commitmentIdToEquip) {
+            if (toEquipId == existingEquippedWearableId && _existingEquippedDepositId == _depositIdToEquip) {
                 continue;
             }
 
             //If a wearable was equipped in this slot and can be transferred, transfer back to owner.
-            
             if (existingEquippedWearableId != 0 && s.itemTypes[existingEquippedWearableId].canBeTransferred) {
-                // Unequip wearable (Sets to 0)
+                // We deleted the wearable from the slot, so we are able to withdraw it back to the diamond or the user
+                // this line necessary to cover the edge case when the wearables to equip and unequip are the same, but the depositId is different
```

Comment:
```suggestion
                // To prevent the function `removeFromParent` to revert, it's necessary first to unequip this Wearable (delete from storage slot)
                // This is an edge case introduced by delegated Wearables, since users can now equip and unequip Wearables of same tokenId (but different depositId)
```